numpy/core/multiarray.py

```diff
@@ -945,15 +945,29 @@ def bincount(x, weights=None, minlength=None):
     >>> np.bincount(x).size == np.amax(x)+1
     True
 
-    The input array needs to be of integer dtype, otherwise a
-    TypeError is raised:
+    The input array needs have an integer dtype smaller or equal to the
+    native Python array index type, otherwise a TypeError is raised:
 
     >>> np.bincount(np.arange(5, dtype=float))
     Traceback (most recent call last):
       ...
     TypeError: Cannot cast array data from dtype('float64') to dtype('int64')
     according to the rule 'safe'
 
+    This also means that the following code will not work on a 32-bit system:
+
+    >>> import sys
+    >>> if sys.maxsize <= 2**32:
+    ...     np.bincount(np.arange(5, dtype=np.int64))
+    ...     # TypeError: Cannot cast array data from dtype('int64') to
+    ...     # dtype('int32') according to the rule 'safe'
```

Comment:
```suggestion
    TypeError: Cannot cast array data from dtype('int64') to
    dtype('int32') according to the rule 'safe'
```
It's standard to show the error itself instead of a comment saying what the error will be.

Reply:
I get that, and it's what I tried at first.
But I found no way to force this to be evaluated in a 32-bit Python context, so I put it like this.
Without it, the "Run Refguide Check" will fail consistently, because all the unit tests are executed on a 64-bit architecture.

Reply:
> Without it, the "Run Refguide Check" will fail consistently, because all the unit tests are executed on a 64-bit architecture.

Oh, that's interesting. In that case, yes, leave it like this.